oasislmf/manager.py

```diff
@@ -417,6 +418,67 @@ def generate_oasis_files(
             exposure_fp, _keys_fp, exposure_profile=exposure_profile
         )
 
+        # Fill summary of exposure data dictionary
+        def fill_exposure_summary(
+            exposure_summary_df,
+            exposure_summary,
+            peril,
+            status,
+        ):
+
+            for coverage_type in SUPPORTED_COVERAGE_TYPES:
+                tiv_sum = exposure_summary_df.loc[
+                    (exposure_summary_df['peril_id']==peril) &\
+                    (exposure_summary_df['coverage_type_id']==SUPPORTED_COVERAGE_TYPES[coverage_type]['id']), 'tiv'].sum()
+                tiv_sum = float(tiv_sum)
+                exposure_summary[peril][status]['tiv_by_coverage'][coverage_type] = tiv_sum
+                if coverage_type in exposure_summary[peril]['all']['tiv_by_coverage']:
+                    exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] += tiv_sum
+                else:
+                    exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] = tiv_sum
+                exposure_summary[peril][status]['tiv'] += tiv_sum
+                exposure_summary[peril]['all']['tiv'] += tiv_sum
+            exposure_summary[peril][status]['location_ids'] =\
+                exposure_summary_df.loc[
+                    exposure_summary_df['peril_id']==peril,
+                    'locnumber'].drop_duplicates().tolist()
+
```

Comment:
@hchagani-oasislmf This method isn't returning or generating anything.

Reply:
Line 441 includes the assignment, split over multiple lines to ease readability:
```
exposure_summary[peril][status]['location_ids'] =\
    exposure_summary_df.loc[
        exposure_summary_df['peril_id']==peril,
        'locnumber'].drop_duplicates().tolist()
```

Reply:
OK, so you're modifying the summary dataframe that the method receives from the main body of `generate_oasis_files` method in the manager. But as I said, I would move this method and the code on lines 465-476 to a separate module in `model_preparation` - probably `summaries.py` which I'm adding in my next update of `develop`. The methods for generating that summary report and/or writing it to file, should be imported in the manager, and then used wherever it is needed.